src/App.js

```diff
@@ -0,0 +1,111 @@
+import DocumentList from "./component/DocumentList.js";
+import Document from "./component/Document.js";
+import { initRouter, push } from "./core/router.js";
+import nestedObjectToArray from "./core/nestedObjectToArray.js";
+import { getItem, setItem } from "./core/storage.js";
+import { request } from "./core/api.js";
+import editor from "./component/Editor.js";
+
+export default function App({ $target }) {
+  this.state = getItem("tempSaveKey") || alert("새로고침을 해주세요");
+
+  this.setState = async (nextState) => {
+    this.state = nextState;
+    await documentList.setState(this.state);
+    await document.setState(this.state);
```

Comment:
```suggestion
    this.state = nextState;
    
    await documentList.setState(this.state);
    await document.setState(this.state);
```

p5: 가독성이 좋은 코드를 고민하시는 것 같아서... 저는 이런 경우 개행을 두는 편입니다!! 훨씬 읽기 편한 것 같더라구요 😆(같은 것끼리 묶기!)

Reply:
단순 궁금증.

setState가 async 하게 동작하는 이유는 무엇인가요?